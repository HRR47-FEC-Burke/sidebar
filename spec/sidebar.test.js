/* eslint no-undef: "error" */
/* global test, expect, describe, it */
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Sidebar from '../sidebar';

const product1 = {
  name: 'Martin 75th Anniversary of Grand Ole Opry, #223 of #650 owned by Brother Oswald from Roy Acuff Band',
  condition: 'mint',
  shippingFee: 79,
  priceOriginal: 3499,
  priceActual: 3299,
  isOpenToOffers: true
};

const seller1 = {
  name: 'Average Joe\'s Guitars',
  address: 'Beaufort, NC, United States',
  isQuickShipper: true,
  joinedYear: 2016,
  reviews: { rating: 4 }
};

const product2 = {
  name: 'Martin D-28 John Lennon Sitka Spruce / Rosewood Dreadnought',
  condition: 'mint',
  shippingFee: 89,
  priceOriginal: 3399,
  priceActual: 3199,
  isOpenToOffers: false
};

const seller2 = {
  name: 'Hitchhiker Music',
  address: 'Busan, Korea, Republic of',
  isQuickShipper: false,
  joinedYear: 2018,
  reviews: { rating: 5 }
};

// jest test
test('testing on the sidebar component', () => {
  const component = renderer.create(
    <Sidebar product={product1} seller={seller1} />
  );

  const page = component.toJSON();
  // Checks if there is any change in the file appearance
  // Doesn't check the inside though
  expect(page).toMatchSnapshot();
});

// enzyme test
configure({ adapter: new Adapter() });

describe('Sidebar', () => {
  const wrapper = shallow(<Sidebar product={product1} seller={seller1} />);

  it('It should exist', () => {
    expect(wrapper.instance()).toBeTruthy();
  });

  it('It should have state component product', () => {
    expect(wrapper.instance().state.product.name).toEqual('Martin 75th Anniversary of Grand Ole Opry, #223 of #650 owned by Brother Oswald from Roy Acuff Band');
    expect(wrapper.instance().state.product.condition).toEqual('mint');
    expect(wrapper.instance().state.product.shippingFee).toEqual(79);
    expect(wrapper.instance().state.product.priceOriginal).toEqual(3499);
    expect(wrapper.instance().state.product.priceActual).toEqual(3299);
    expect(wrapper.instance().state.product.isOpenToOffers).toEqual(true);
  });

  it('It should have state component seller', () => {
    expect(wrapper.instance().state.seller.name).toEqual('Average Joe\'s Guitars');
    expect(wrapper.instance().state.seller.address).toEqual('Beaufort, NC, United States');
    expect(wrapper.instance().state.seller.isQuickShipper).toEqual(true);
    expect(wrapper.instance().state.seller.joinedYear).toEqual(2016);
    expect(wrapper.instance().state.seller.reviews.rating).toEqual(4);
  });

  it('It should have clickable button', () => {
    const watchButton = wrapper.find('#watchButton');
    expect(watchButton.text()).toEqual('â˜† Watch');
    // clicking button causes error in test, but run fine on web page
    expect(() => { watchButton.simulate('click'); }).toThrow(Error);
    wrapper.instance().setState({ isWatched: true });
    expect(() => { watchButton.simulate('click'); }).toThrow(Error);
  });

  it('It should work with different input data', () => {
    const wrapper2 = shallow(<Sidebar product={product2} seller={seller2} />);
    expect(wrapper2.instance().state.product.name).toEqual('Martin D-28 John Lennon Sitka Spruce / Rosewood Dreadnought');
    expect(wrapper2.instance().state.seller.name).toEqual('Hitchhiker Music');
  });
});
