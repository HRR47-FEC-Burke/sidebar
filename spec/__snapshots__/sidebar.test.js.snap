// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checks if there is any change in the file appearance when it renders 1`] = `
<div
  className="sb-whole"
>
  <div
    className="sb-bigText"
  >
    Martin 75th Anniversary of Grand Ole Opry, #223 of #650 owned by Brother Oswald from Roy Acuff Band
  </div>
  <div
    className="sb-smallText sb-green sb-dashBottomBorder"
  >
    mint
  </div>
  <div
    className="sb-bigSpace"
  >
    <div
      className="sb-smallText"
    >
      Shipping fee: 
      79
    </div>
    <div
      className="sb-lineThroughText"
    >
      $
      3499
    </div>
    <div
      className="sb-bold sb-bigText"
    >
      $
      3299
    </div>
    <div
      className="sb-smallText sb-grey"
    >
      +Shipping
    </div>
  </div>
  <button
    className="sb-bigButton"
  >
    Add to Cart
  </button>
  <div>
    <button
      className="sb-smallButton"
    >
      Make an Offer
    </button>
    <button
      className="sb-smallButton sb-floatRight"
      id="watchButton"
      onClick={[Function]}
    >
      ☆ Watch
    </button>
  </div>
  <div
    className="sb-smallText sb-goldenrod sb-dialogBubble"
  >
    This seller is open to offers
  </div>
  <div
    className="sb-greyBottomBorder"
  >
    <div
      className="sb-bigSpace sb-floatLeft"
    >
      <img
        alt=""
        src="test-file-stub"
      />
    </div>
    <div
      className="sb-bigSpace"
    >
      <div
        className="sb-smallText sb-bold"
      >
        Still Shipping Quickly
      </div>
      <div
        className="sb-smallText sb-grey"
      >
        This seller is shipping orders within 24 hours, on average.
        <br />
        <br />
      </div>
    </div>
  </div>
  <div
    className="sb-greyBottomBorder"
  >
    <div
      className="sb-bigSpace sb-floatLeft"
    >
      <img
        alt=""
        src="test-file-stub"
      />
    </div>
    <div
      className="sb-bigSpace"
    >
      <div
        className="sb-smallText sb-bold"
      >
        Buy With Confidence
      </div>
      <div
        className="sb-smallText sb-grey"
      >
        Reverb Protection has you covered. We provide a safe community for finding the gear you want
      </div>
    </div>
  </div>
  <section
    className="sb-smallText"
  >
    <div
      className="sb-bigSpace sb-grey sb-half sb-floatLeft"
    >
      Shipped From
      <div
        className="sb-bold sb-black"
      >
        Average Joe's Guitars
      </div>
      <div>
        Beaufort, NC, United States
      </div>
      <div
        className="sb-orange"
      >
        ★★★★☆
      </div>
      <div>
        <div>
          Joined Reverb
        </div>
        <div>
          2016
        </div>
      </div>
    </div>
    <div
      className="sb-extraSpace sb-blue"
    >
      <img
        alt=""
        src="test-file-stub"
      />
    </div>
  </section>
  <div>
    <button
      className="sb-smallButton sb-smallText"
    >
      Message Seller
    </button>
    <button
      className="sb-smallButton sb-smallText sb-floatRight"
    >
      Payment and Returns
    </button>
  </div>
</div>
`;
